src/core.ts

```diff
@@ -404,6 +404,16 @@ export const TrackingConsentWithNulls = t.record(
 /** Type override */
 export type TrackingConsentWithNulls = t.TypeOf<typeof TrackingConsent>;
 
+const TCFReservedMetadata = t.partial({
+  tcString: t.string,
+});
+
+const ReservedMetadata = t.partial({
+  tcmp: t.partial({
```

Comment:
I'm not the most update to date but what does `tcmp` represent?

Reply:
So we're storing some "reserved" metadata that we don't want customers to access, and rather than putting them as top level keys where we might have to worry about collisions our communicating to customers to avoid these keywords, we're grouping them all under a `tcmp` field to shrink the chance of collision as much as possible

Reply:
So a `tcmp` field is just a nested field to sorta hide these values that we don't want the customer to see?

Reply:
Yep, exactly! Then the values themselves are just arbitrary module data that we want synced